media_driver/linux/common/ddi/media_libva.cpp

```diff
@@ -3664,95 +3664,50 @@ static VAStatus DdiMedia_SyncSurface (
+    DDI_FUNCTION_ENTER();
 
-        if (tableLen > 0 && surface->curStatusReportQueryState == DDI_MEDIA_STATUS_REPORT_QUERY_STATE_PENDING)
-        {
-            // Query the status for all of surfaces which have finished
-            for(i = 0; i < tableLen; i++)
-            {
-                MOS_ZeroMemory(&tempVpReport, sizeof(QUERY_STATUS_REPORT_APP));
-                vpCtx->pVpHal->GetStatusReport(&tempVpReport, 1);
+    DDI_CHK_NULL(ctx,    "nullptr ctx",    VA_STATUS_ERROR_INVALID_CONTEXT);
 
-                // StatusFeedBackID is last time submitted Target Surface ID which is set in BeginPicture,
-                // So we can know the report is for which surface here.
-                DDI_MEDIA_SURFACE *tempSurface = DdiMedia_GetSurfaceFromVASurfaceID(mediaCtx, tempVpReport.StatusFeedBackID);
-                if(tempSurface == nullptr)
-                {
-                    return VA_STATUS_ERROR_OPERATION_FAILED;
-                }
+    PDDI_MEDIA_CONTEXT mediaCtx = DdiMedia_GetMediaContext(ctx);
+    DDI_CHK_NULL(mediaCtx,               "nullptr mediaCtx",               VA_STATUS_ERROR_INVALID_CONTEXT);
+    DDI_CHK_NULL(mediaCtx->pSurfaceHeap, "nullptr mediaCtx->pSurfaceHeap", VA_STATUS_ERROR_INVALID_CONTEXT);
 
-                // Update the status of the surface which is reported.
-                tempSurface->curStatusReport.vpp.status = (uint32_t)tempVpReport.dwStatus;
-                tempSurface->curStatusReportQueryState  = DDI_MEDIA_STATUS_REPORT_QUERY_STATE_COMPLETED;
+    DDI_CHK_LESS((uint32_t)surface_id, mediaCtx->pSurfaceHeap->uiAllocatedHeapElements, "Invalid render_target", VA_STATUS_ERROR_INVALID_SURFACE);
 
-                if(tempVpReport.StatusFeedBackID == render_target)
-                {
-                    break;
-                }
-            }
-        }
+    DDI_MEDIA_SURFACE  *surface = DdiMedia_GetSurfaceFromVASurfaceID(mediaCtx, surface_id);
+    DDI_CHK_NULL(surface,    "nullptr surface",      VA_STATUS_ERROR_INVALID_CONTEXT);
+    if (surface->pCurrentFrameSemaphore)
+    {
+        DdiMediaUtil_WaitSemaphore(surface->pCurrentFrameSemaphore);
+        DdiMediaUtil_PostSemaphore(surface->pCurrentFrameSemaphore);
+    }
 
-        if (surface->curStatusReportQueryState == DDI_MEDIA_STATUS_REPORT_QUERY_STATE_COMPLETED)
-        {
-            if(surface->curStatusReport.vpp.status == VPREP_OK)
-            {
-                return VA_STATUS_SUCCESS;
-            }
-            else if(surface->curStatusReport.vpp.status == VPREP_NOTREADY)
-            {
-                return VA_STATUS_ERROR_HW_BUSY;
-            }
-            else
-            {
-                return VA_STATUS_ERROR_OPERATION_FAILED;
-            }
-        }
-        else
-        {
-            return VA_STATUS_ERROR_OPERATION_FAILED;
-        }
+    // zero is a expected return value when not hit timeout
+    int64_t timeoutBoWait = timeout_ns >= DDI_MAX_TIMEOUT ? DDI_INFINITE_TIMEOUT : (int64_t)timeout_ns;
```

Comment:
Yuchen,
I see you made it the way if value of uint64_t type timeout is  higher than max_vaue(int64_t) you do wait call with infinite. Is there a reason for it instead of calling mos_gem_bo_wait two times, first with max_vaue(int64_t), second with "orig_time_out - max_vaue(int64_t)". 
I don't insist on the 2nd solution, just want to understand objective reasons of following the proposed solution.

Reply:
because this is to align with i915 bo_wait behavior (it's int64_t and when <0, infinite wait)
when the value > max_value(int64_t), the bit 63 will be 1.  When uint64_t max bit is 1, int64_t <0. So actual range is 0-62 bit for timeout.Then bit63 set to 1 means infinite.

We can use option2. But we need firstly allign which value of uint64_t is indicating VA_TIMEOUT_INFINITE   (maybe max(uint64_t) ?)

Reply:
>We can use option2. But we need firstly allign which value of uint64_t is indicating VA_TIMEOUT_INFINITE (maybe max(uint64_t) ?)

+1 for this option